ArduCopter/mode_guided.cpp

```diff
@@ -84,8 +84,27 @@ void ModeGuided::run()
     case SubMode::Angle:
         angle_control_run();
         break;
+
+    case SubMode::Circle:
+        switch (loiter_turns.get_state()) {
+        case _LoiterTurns::State::CIRCLING: {
+            loiter_turns.circle_run();
+            break;
+        }
+        case _LoiterTurns::State::MOVING_TO_EDGE:
+            // use waypoint navigation controller in this state
+            wp_control_run();
+            break;
+        case _LoiterTurns::State::READY:
+            break;
+        }
+
+        if (loiter_turns.is_completed()) {
+            // switch back to WP mode once we complete loiter turns
+            guided_mode = SubMode::WP;
```

Comment:
Does this need any initialisation?

Reply:
Thanks for bringing this to my notice. I have initialised in `is_completed()` when we return true.